browser/extensions/ethereum_remote_client_unittest.cc

```diff
@@ -154,3 +164,18 @@ TEST_F(BraveWalletUnitTest, TestSaveToPrefs) {
   ASSERT_EQ(GetPrefs()->GetString(kERCEncryptedSeed),
             "IQu5fUMbXG6E7v8ITwcIKL3TI3rst0LU1US7ZxCKpgAGgLNAN6DbCN7nMF2Eg7Kx");
 }
+
+TEST_F(BraveWalletUnitTest, MaybeLoadCryptoWalletsExtension) {
+  auto* profile = ProfileManager::GetActiveUserProfile();
+  std::unique_ptr<EthereumRemoteClientService> service(
+      new EthereumRemoteClientService(
+          profile, std::make_unique<MockEthereumRemoteClientDelegateImpl>()));
+  base::RunLoop loop;
+  bool callback_called = false;
+  service->MaybeLoadCryptoWalletsExtension(
+      base::BindLambdaForTesting([&]() { callback_called = true; }));
+  ASSERT_TRUE(service->load_ui_callback_);
+  service->MaybeLoadCryptoWalletsExtension(base::DoNothing());
+  std::move(service->load_ui_callback_).Run();
```

Comment:
This test is completely wrong.
We should have two consecutive call of `MaybeLoadCryptoWalletsExtension` with different callbacks and call `CryptoWalletsExtensionReady()` afterward
```
MaybeLoadCryptoWalletsExtension(callback1)
MaybeLoadCryptoWalletsExtension(callback2)
CryptoWalletsExtensionReady()
```
and make sure both callbacks are ran.
hence you don't need to access private `load_ui_callback_`

Reply:
thanks! updated